Re-enable pytest-xdist for Windows

The pytest-xdist mode works fine on my Windows machine but fails for inscrutable reasons on GH Windows 2022 runners like so:
https://github.com/a-scie/lift/actions/runs/4974563303/jobs/8902239729#step:6:106

The only evidence of error is an exit code of 1 with no stderr:
```
nox > pytest -n auto -vvs
============================= test session starts =============================
platform win32 -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0 -- D:\a\lift\lift\.nox\test-3-11\Scripts\python.EXE
cachedir: .pytest_cache
rootdir: D:\a\lift\lift
configfile: pyproject.toml
plugins: anyio-3.6.2, xdist-3.3.0
created: 2/2 workers
2 workers [9 items]

scheduling tests via LoadScheduling

tests/test_a_scie.py::test_ptex_latest 
tests/test_config.py::test_parse 
[gw1] PASSED tests/test_config.py::test_parse 
tests/test_config.py::test_interpreter_groups 
  0%|          | 0.00/1.36M [00:00<?, ?B/s]
100%|##########| 1.36M/1.36M [00:00<00:00, 91.3MB/s]

[gw0] PASSED tests/test_a_scie.py::test_ptex_latest 
tests/test_a_scie.py::test_ptex_version 
  0%|          | 0.00/1.33M [00:00<?, ?B/s]
100%|##########| 1.33M/1.33M [00:00<00:00, 44.6MB/s]

[gw0] PASSED tests/test_a_scie.py::test_ptex_version 
tests/test_exe.py::test_hash 
[gw1] FAILED tests/test_config.py::test_interpreter_groups 
tests/test_exe.py::test_use_platform_suffix 
[gw1] PASSED tests/test_exe.py::test_use_platform_suffix 
tests/test_exe.py::test_nested_filenames[https://github.com/a-scie/lift/issues/2] 
[gw0] PASSED tests/test_exe.py::test_hash 
tests/test_exe.py::test_dogfood 
    
        If check is True and the exit code was non-zero, it raises a
        CalledProcessError. The CalledProcessError object will have the return code
        in the returncode attribute, and output & stderr attributes if those streams
        were captured.
    
        If timeout is given, and the process takes too long, a TimeoutExpired
        exception will be raised.
    
        There is an optional argument "input", allowing you to
        pass bytes or a string to the subprocess's stdin.  If you use this argument
        you may not also use the Popen constructor's "stdin" argument, as
        it will be used internally.
    
        By default, all communication is in bytes, and therefore any "input" should
        be bytes, and the stdout and stderr will be bytes. If in text mode, any
        "input" should be a string, and stdout and stderr will be strings decoded
        according to locale encoding, or by "encoding" if set. Text mode is
        triggered by setting any of text, encoding, errors or universal_newlines.
    
        The other arguments are the same as for the Popen constructor.
        """
        if input is not None:
            if kwargs.get('stdin') is not None:
                raise ValueError('stdin and input arguments may not both be used.')
            kwargs['stdin'] = PIPE
    
        if capture_output:
            if kwargs.get('stdout') is not None or kwargs.get('stderr') is not None:
                raise ValueError('stdout and stderr arguments may not be used '
                                 'with capture_output.')
            kwargs['stdout'] = PIPE
            kwargs['stderr'] = PIPE
    
        with Popen(*popenargs, **kwargs) as process:
            try:
                stdout, stderr = process.communicate(input, timeout=timeout)
            except TimeoutExpired as exc:
                process.kill()
                if _mswindows:
                    # Windows accumulates the output in a single blocking
                    # read() call run on child threads, with the timeout
                    # being done in a join() on those threads.  communicate()
                    # _after_ kill() is required to collect that and add it
                    # to the exception.
                    exc.stdout, exc.stderr = process.communicate()
                else:
                    # POSIX _communicate already populated the output so
                    # far into the TimeoutExpired exception.
                    process.wait()
                raise
            except:  # Including KeyboardInterrupt, communicate handled that.
                process.kill()
                # We don't call process.wait() as .__exit__ does that for us.
                raise
            retcode = process.poll()
            if check and retcode:
>               raise CalledProcessError(retcode, process.args,
                                         output=stdout, stderr=stderr)
E               subprocess.CalledProcessError: Command '[WindowsPath('D:/a/lift/lift/.nox/test-3-11/tmp/pytest-of-runneradmin/pytest-0/popen-gw1/test_interpreter_groups0/igs.exe')]' returned non-zero exit status 1.

C:\hostedtoolcache\windows\Python\3.11.3\x64\Lib\subprocess.py:571: CalledProcessError
=========================== short test summary info ===========================
FAILED tests/test_config.py::test_interpreter_groups - subprocess.CalledProcessError: Command '[WindowsPath('D:/a/lift/lift/.nox/test-3-11/tmp/pytest-of-runneradmin/pytest-0/popen-gw1/test_interpreter_groups0/igs.exe')]' returned non-zero exit status 1.
======================== 1 failed, 8 passed in 16.25s =========================
nox > Command pytest -n auto -vvs failed with exit code 1
nox > Session test-3.11 failed.
Error: Process completed with exit code 1.
```

This seems to indicate a Windows-specific bug for some Windows configurations.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -36,6 +36,10 @@ jobs:
         uses: actions/checkout@v3
       - name: Check Formatting & Lints
         run: nox -e lint
+      - name: Configure Windows pytest short tmp dir path
+        if: matrix.os == 'windows-2022'
+        run: |
+          echo PYTEST_ADDOPTS="--basetemp C:\\tmp\\pytest" >> ${GITHUB_ENV}
       - name: Unit Tests
         run: nox -e test -- -vvs
       - name: Build & Package

--- a/lift.toml
+++ b/lift.toml
@@ -5,8 +5,8 @@ description = "Ship your interpreted executables using science."
 [[lift.interpreters]]
 id = "cpython"
 provider = "PythonBuildStandalone"
-release = "20230116"
-version = "3.11.1"
+release = "20230507"
+version = "3.11.3"
 lazy = true
 
 [[lift.files]]

--- a/noxfile.py
+++ b/noxfile.py
@@ -253,12 +253,7 @@ def create_zipapp(session: Session) -> Path:
 def test(session: Session) -> None:
     science_pyz = create_zipapp(session)
     test_env = {"BUILD_ROOT": str(BUILD_ROOT), "SCIENCE_TEST_PYZ_PATH": str(science_pyz)}
-
-    # TODO(John Sirois): Figure out why this fails for GH Windows 2022 runners and re-enable for
-    #  Windows: https://github.com/a-scie/lift/issues/9
-    xdist_args = [] if IS_WINDOWS else ["-n" "auto"]
-
-    session.run("pytest", *xdist_args, *(session.posargs or ["-v"]), env=test_env)
+    session.run("pytest", "-n" "auto", *(session.posargs or ["-v"]), env=test_env)
 
 
 @python_session()

--- a/tests/data/interpreter-groups.toml
+++ b/tests/data/interpreter-groups.toml
@@ -8,14 +8,12 @@ version = "0.11.0"
 [[lift.interpreters]]
 id = "cpython310"
 provider = "PythonBuildStandalone"
-release = "20230116"
 version = "3.10"
 lazy = true
 
 [[lift.interpreters]]
 id = "cpython311"
 provider = "PythonBuildStandalone"
-release = "20230116"
 version = "3.11"
 lazy = true
 

--- a/tests/test_exe.py
+++ b/tests/test_exe.py
@@ -7,6 +7,7 @@ import io
 import itertools
 import os
 import re
+import shutil
 import subprocess
 import sys
 from dataclasses import dataclass
@@ -159,7 +160,7 @@ def test_nested_filenames(
     dist_dir.mkdir()
 
     dest = dist_dir / science_pyz.name
-    os.link(science_pyz, dest)
+    shutil.copy(science_pyz, dest)
 
     with config.open(mode="r") as fp:
         config_data = toml.load(fp)

